### ISSUE TITLE
Problem: PID = nil when global:true and both nodes online

### ISSUE THREAD

**Issue by AnderLuiz** [c-rack/quantum-elixir#180]
I have 2 nodes. when only one node is up everything works fine, but when both nodes are up, Node two only add jobs with PID=nil.

my config file:
`config :quantum,
   global?: true,
   timeout: 30_000,
   timezone: "America/Sao_Paulo"`

Quantum version: `1.9.1`

I am adding this task on both nodes for testing

`task_name = “test_task”
job = %Quantum.Job{schedule: "58 20 17 4 *", task: fn -> 
IO.puts “testing”  
Quantum.delete_job(task_name)
end }
Quantum.add_job(task_name, job)`

On node 1, the result of `Quantum.jobs` is:

`[{"test_task",
  %Quantum.Job{args: [], name: "test_task", nodes: [:"node1@ip_address"],
   overlap: true, pid: #PID<0.8592.0>, schedule: ~e[58 14 17 4 * *],
   state: :active, task: #Function<20.52032458/0 in :erl_eval.expr/5>,
   timezone: "America/Sao_Paulo"}},...`

If I add same job on Node 2:

```
[{"test_task",
  %Quantum.Job{args: [], name: "test_task", nodes: [:"node2@ip_address"],
   overlap: true, pid: nil, schedule: ~e[58 20 17 4 * *], state: :active,
   task: #Function<20.52032458/0 in :erl_eval.expr/5>,
   timezone: "America/Sao_Paulo"}},...
```

All tasks when node is `node2@ip_address` contains `pid: nil` and does not work just when both nodes are online.

The result of `Node.list` is ok, and when I add a job on a node, both of them can see the changes.

Edit: Debugging the code, when the nodes  are connected, `Quantum.run(state)` function is only being called at node1. So only the jobs destined to node1 are being executed. 
This change in job.ex solves my problem, but I think it is not the best solution.
  ```
def executable?(job) do
    cond do
      job.state != :active    -> false # Do not execute inactive jobs
      (not node() in job.nodes) and not Application.get_env(:quantum, :global?, false)   -> false # <<<<<<<This line here
      job.overlap == true     -> true  # Job may overlap
      job.pid == nil          -> true  # Job has not been started
      is_alive?(job.pid)      -> false # Previous job is still running
      true                    -> true  # Previous job has finished
    end
  end

```

**Comment by maennchen** [comment]
I propose that the `Quantum.Runner` only runs one if `global?` is true.

Should the runner a) start a task for each node in `job.nodes` or should it b) run it only on one server?

If a), we need to save a map of `%{node_name => pid}` instead of just one pid.
Also it gets more complicated to determine overlap, if multiple pids are stored.

If we're already on it, should we also add `:any` and `:all` as valid values of `job.nodes` where `:all` runs on all nodes and `:any` runs on just one random node?

@c-rack What do you think?

**Comment by c-rack** [comment]
Option (b) sounds reasonable, also the additions for `job.nodes`

**Comment by maennchen** [comment]
Solved via #185
